src/structure/grid.ts

```diff
@@ -183,6 +160,12 @@ export class ViewersGrid {
         this.oncreate = () => {};
         this.activeChanged = () => {};
 
+        if (maxWidgets !== undefined) {
+            this._maxWidgets = maxWidgets;
```

Comment:
I would validate that maxWidget is lower than 9 here, since this class could be used directly, without going through the validation code in `index.ts`